### PR TITLE
Rename diagram_cache to diagram_image_cache

### LINKED ISSUE #247: Rename "diagram cache" to "diagram image cache"
To make it less ambiguous what the feature is about, we should explicitly mention that it's about the graphical representations of the diagrams, i.e. images.

This includes the main entry point method `update_diagram_cache`, which constitutes a breaking API change. Therefore there will be a deprecation period.

### PR DESCRIPTION
**Pull request by dahbar**
Closes #247

Added a deprecation warning for `diagram_cache` and `diagram_cache_subdir` in the MelodyLoader and for the `update_diagram_cache` function.

### CODE REVIEW

File: capellambse/model/__init__.py [hunk 1]
```diff
@@ -238,13 +250,31 @@ def __init__(
...
             warnings.warn(
-                (
-                    "The 'jupyter_untrusted' argument is no longer needed and"
-                    " will be removed soon. Please remove it from your calls."
-                ),
+                "The 'jupyter_untrusted' argument is no longer needed and"
+                " will be removed soon. Please remove it from your calls.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
```
Comment (ewuerger) [inline capellambse/model/__init__.py:257]: Has nothing to do with the `diagram_cache` rename but still good catch. Mixing these things into one commit can be annoying in the future. Keep that in mind. Even though, I forget this from time to time,...

**Review by ewuerger — APPROVED** [review]

File: capellambse/model/__init__.py (hunk 1 above)
Comment (Wuestengecko) [reply capellambse/model/__init__.py:257]: What's the "good catch"? As far as I can see, the only thing that changed was some formatting (no parenthesis around the string anymore).

Comment (dahbar) [reply capellambse/model/__init__.py:257]: I believe these formatting changes were made by running `black --preview capellambse/model/__init__.py` and not really by me.

**Comment by Wuestengecko** [comment]
We determined that we won't be doing this after all. The name "diagram cache" has already been established too well, and is also used in other places and projects (like [DSD-DBS/capella-collab-manager](https://github.com/DSD-DBS/capella-collab-manager)), changing it now would just cause unnecessary headache downstream.

**Pull request closed without merging**